app/assets/javascripts/controllers/provider_foreman/provider_foreman_form_controller.js

```diff
@@ -1,110 +1,110 @@
-    var init = function() {
-      $scope.providerForemanModel = {
-        provtype: '',
-        name: '',
-        url: '',
-        zone: '',
-        verify_ssl: '',
-        log_userid: '',
-        log_password: '',
-        log_verify: ''
-      };
-      $scope.formId = providerForemanFormId;
-      $scope.afterGet = false;
-      $scope.validateClicked = miqService.validateWithAjax;
-      $scope.modelCopy = angular.copy( $scope.providerForemanModel );
-      $scope.model = 'providerForemanModel';
-
-      ManageIQ.angular.scope = $scope;
-
-      if (providerForemanFormId == 'new') {
-        $scope.newRecord = true;
-
-        $http.get('/provider_foreman/provider_foreman_form_fields/' + providerForemanFormId)
-          .then(getProviderForemanFormData)
-          .catch(miqService.handleFailure);
-      } else {
-        $scope.newRecord = false;
-
-        miqService.sparkleOn();
-
-        $http.get('/provider_foreman/provider_foreman_form_fields/' + providerForemanFormId)
-          .then(getProviderForemanFormData)
-          .catch(miqService.handleFailure);
-      }
-    };
-
-    $scope.canValidateBasicInfo = function () {
-      if ($scope.isBasicInfoValid())
-        return true;
-      else
-        return false;
+  var vm = this;
+
+  vm.providerForemanModel = {
+    provtype: '',
+    name: '',
+    url: '',
+    zone: '',
+    verify_ssl: '',
+    log_userid: '',
+    log_password: '',
+    log_verify: ''
+  };
+
+  vm.formId = providerForemanFormId;
+  vm.afterGet = false;
+  vm.validateClicked = miqService.validateWithAjax;
+  vm.modelCopy = angular.copy( vm.providerForemanModel );
+  vm.model = 'providerForemanModel';
+
+  ManageIQ.angular.scope = vm;
+
+
+  if (providerForemanFormId == 'new') {
+    vm.newRecord = true;
+
+    $http.get('/provider_foreman/provider_foreman_form_fields/' + providerForemanFormId)
+      .then(getProviderForemanFormData)
+      .catch(miqService.handleFailure);
+  } else {
+    vm.newRecord = false;
+
+    miqService.sparkleOn();
+
+    $http.get('/provider_foreman/provider_foreman_form_fields/' + providerForemanFormId)
+      .then(getProviderForemanFormData)
+      .catch(miqService.handleFailure);
+  }
+
+  vm.canValidateBasicInfo = function () {
+    if (vm.isBasicInfoValid())
+      return true;
+    else
+      return false;
+  }
+
+  vm.isBasicInfoValid = function(angularForm) {
+    if(angularForm.url.$valid &&
+      angularForm.log_userid.$valid &&
+      angularForm.log_password.$valid &&
+      angularForm.log_verify.$valid)
+      return true;
+    else
+      return false;
+  };
+
+  var providerForemanEditButtonClicked = function(buttonName, serializeFields) {
+    miqService.sparkleOn();
+    var url = '/provider_foreman/edit/' + providerForemanFormId + '?button=' + buttonName;
+    if (serializeFields === undefined) {
+      miqService.miqAjaxButton(url);
+    } else {
+      miqService.miqAjaxButton(url, serializeFields);
```

Comment:
since you're rewriting this controller anyway, please replace `miqAjaxButton(url, true)` with `miqAjaxButton(url, vm.providerForemanModel)`